napari/_qt/qt_main_window.py

```diff
@@ -1550,27 +1556,61 @@ def _screenshot(
         flash : bool
             Flag to indicate whether flash animation should be shown after
             the screenshot was captured.
-        size : tuple (int, int)
+        size : tuple of two ints, optional
             Size (resolution height x width) of the screenshot. By default, the currently displayed size.
             Only used if `canvas_only` is True.
-        scale : float
+        scale : float, optional
             Scale factor used to increase resolution of canvas for the screenshot.
             By default, the currently displayed resolution.
             Only used if `canvas_only` is True.
         canvas_only : bool
             If True, screenshot shows only the image display canvas, and
             if False include the napari viewer frame in the screenshot,
             By default, True.
+        fit_to_data_extent: bool
+            Tightly fit the canvas around the data to prevent margins from
+            showing in the screenshot. If False, a screenshot of the whole
+            currently visible canvas will be generated.
 
         Returns
         -------
         img : QImage
         """
         from napari._qt.utils import add_flash_animation
 
+        canvas = self._qt_viewer.canvas
+        prev_size = canvas.size
+        camera = self._qt_viewer.viewer.camera
+        old_center = camera.center
+        old_zoom = camera.zoom
+        if fit_to_data_extent:
+            if not canvas_only:
+                raise ValueError(
+                    trans._(
+                        "'fit_to_data_extent' cannot be set to True if 'canvas_only' is"
+                        ' set to False',
+                        deferred=True,
+                    )
+                )
+            ndisplay = self._qt_viewer.viewer.dims.ndisplay
+            if ndisplay > 2:
+                raise NotImplementedError(
+                    trans._(
+                        'fit_to_data_extent=True is not yet implemented for 3D. '
+                        'Please set fit_to_data_extent to False in 3D view.',
+                        deferred=True,
+                    )
+                )
+
+            self._qt_viewer.viewer.reset_view()
+            extent_world = self._qt_viewer.viewer.layers.extent.world[1][
+                -ndisplay:
+            ]
+            extent_step = self._qt_viewer.viewer.layers.extent.step[-ndisplay:]
```

Comment:
After rethink, I think that we should use here min
```suggestion
            extent_step = min(self._qt_viewer.viewer.layers.extent.step[-ndisplay:])
```
Otherwise, if someone uses non-equal scale in x,y then the proportion of produced image will be wrong.